.travis.yml

```diff
@@ -11,5 +11,7 @@ script:
   - if [[ "$TRAVIS_NIGHTLY" == "true" ]]; then cargo test --verbose --all-features; fi
   - if [[ "$TRAVIS_NIGHTLY" == "true" ]]; then cargo bench --verbose --all-features --no-run; fi
   - if [[ "$TRAVIS_NIGHTLY" != "true" ]]; then cargo test --verbose; fi
+  - if [[ "$TRAVIS_NIGHTLY" != "true" ]]; then rustup target add thumbv7m-none-eabi; fi
+  - if [[ "$TRAVIS_NIGHTLY" != "true" ]]; then cargo build --verbose --no-default-features --features=alloc --target thumbv7m-none-eabi; fi
```

Comment:
Did you locally test if this passes tests, while building is great, it is pretty easy to have things failing still. Ideally we would use `cross` to actually run the tests 
Something like

```
cross test --no-default-features --features alloc --target thumbv7m-none-eabi
```

should work if we can get cross to work on travis

Reply:
I might suggest trying to switch to GitHub Actions first before you do this.

Namely [`actions-rs` supports relatively painless cross-compilation](https://github.com/actions-rs/cargo#cross-compilation) with `use_cross: true`.

Reply:
I agree moving to github actions seems ideal, but it should be done in a separate PR IMO. I'll try to open a gh action migration PR this week end.

Reply:
github actions pr is merged

Reply:
Right so this is fun. Cross can't run tests for thumbv7m-none-eabi, since that target doesn't have libstd or libtest. I could build and test for the arm-linux-gnu target, but we still need to have a build step for thumbv7-none-eabi to avoid having accidental dependencies on libstd.

Reply:
So I changed it up by building for thumb, and testing on arm.